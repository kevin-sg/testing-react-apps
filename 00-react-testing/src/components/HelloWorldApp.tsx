import type { FC } from "react";

export const HelloWorldApp: FC = () => {
  return <h1>Hello World App</h1>;
};
