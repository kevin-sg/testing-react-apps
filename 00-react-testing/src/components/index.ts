export * from "./CounterApp";
export * from "./FirstApp";
export * from "./HelloWorldApp";
