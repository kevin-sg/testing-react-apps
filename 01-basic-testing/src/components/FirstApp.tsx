import type { FC } from "react";

interface IFirstAppProps {
  title: string;
  subTitle?: string;
  name?: string;
}

export const FirstApp: FC<IFirstAppProps> = (props) => {
  const { title, subTitle = "Not subtitle", name = "kevin sg" } = props;
  return (
    <>
      <h1 data-testid="test-title"> {title} </h1>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};
