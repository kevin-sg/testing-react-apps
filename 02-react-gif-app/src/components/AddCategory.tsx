import { ChangeEvent, FC, FormEvent, useState } from "react";

interface IAddCategoryProps {
  onNewCategory: (value: string) => void;
}

export const AddCategory: FC<IAddCategoryProps> = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }: ChangeEvent<HTMLInputElement>) => {
    setInputValue(target.value);
  };

  const onSubmit = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();

    if (!inputValue.trim().length) return;

    // setCategories( categories => [ inputValue, ...categories ]);
    setInputValue("");
    onNewCategory(inputValue.trim());
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input type="text" placeholder="Buscar gifs" value={inputValue} onChange={onInputChange} />
    </form>
  );
};
