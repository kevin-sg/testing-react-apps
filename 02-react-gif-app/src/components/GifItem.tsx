import type { FC } from "react";

interface IGifItemProps {
  title: string;
  url: string;
  id: string;
}

export const GifItem: FC<IGifItemProps> = ({ title, url, id }) => {
  return (
    <div className="card">
      <img aria-label={title + "-image"} src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};
