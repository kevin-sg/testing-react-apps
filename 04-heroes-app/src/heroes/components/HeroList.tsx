import { useMemo } from "react";

import { HeroCard } from "./";
import { getHeroesByPublisher } from "../helpers";

interface IHeroListProps {
  publisher: string;
}

export const HeroList: React.FC<IHeroListProps> = ({ publisher }) => {
  const heroes = useMemo(() => getHeroesByPublisher(publisher), [publisher]);

  return (
    <div className="row rows-cols-1 row-cols-md-3 g-3">
      {heroes.map((hero) => (
        <HeroCard key={hero.id} {...hero} />
      ))}
    </div>
  );
};
