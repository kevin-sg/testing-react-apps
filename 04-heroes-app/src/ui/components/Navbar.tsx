import { Link, NavLink, useNavigate } from "react-router-dom";
import { useContext } from "react";

import { AuthContext } from "@/auth";

export const Navbar: React.FC = () => {
  const { authState, logout } = useContext(AuthContext);
  const navigate = useNavigate();

  const onLogout = () => {
    logout();
    navigate("/login", { replace: true });
  };

  return (
    <nav data-testid="test-navbar" className="navbar navbar-expand-sm navbar-dark bg-dark p-2">
      <Link className="navbar-brand" to="/">
        Asociaciones
      </Link>

      <div className="navbar-collapse">
        <div className="navbar-nav">
          <NavLink className={({ isActive }) => `nav-item nav-link  ${isActive ? "active" : ""}`} to="/marvel">
            Marvel
          </NavLink>

          <NavLink className={({ isActive }) => `nav-item nav-link  ${isActive ? "active" : ""}`} to="/dc">
            DC
          </NavLink>

          <NavLink className={({ isActive }) => `nav-item nav-link  ${isActive ? "active" : ""}`} to="/search">
            Search
          </NavLink>
        </div>
      </div>

      <div className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end">
        <ul className="navbar-nav ml-auto">
          <span data-testid="test-username-span" className="nav-item nav-link text-primary">
            {authState.user?.name}
          </span>

          <button data-testid="on-button-click" className="nav-item nav-link btn" onClick={onLogout}>
            Logout
          </button>
        </ul>
      </div>
    </nav>
  );
};
