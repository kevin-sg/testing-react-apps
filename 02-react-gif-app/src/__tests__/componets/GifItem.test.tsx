import { render, screen } from "@testing-library/react";

import { GifItem } from "@/components";

const testData = {
  id: "1",
  title: "Luffy",
  url: "https://www.xtrafondos.com/wallpapers/vertical/monkey-d-luffy-de-one-piece-4017.jpg",
};

describe("Test in <GifItem/>", () => {
  test("should do math with snapshot", () => {
    const { container } = render(<GifItem id={testData.id} title={testData.title} url={testData.url} />);
    // screen.debug();
    expect(container).toMatchSnapshot();
  });

  test("should show the image URl and indicated ALT", () => {
    render(<GifItem id={testData.id} title={testData.title} url={testData.url} />);
    // screen.debug();
    const imageContainer = screen.getByRole("img", { name: testData.title + "-image" });
    // expect(imageContainer).toHaveAttribute("src", testData.url);
    // expect(imageContainer).toHaveAttribute("alt", testData.title + "-");
    expect(imageContainer.getAttribute("src")).toEqual(testData.url);
    expect(imageContainer.getAttribute("alt")).toEqual(testData.title);
  });

  test("should show title in component", () => {
    render(<GifItem id={testData.id} title={testData.title} url={testData.url} />);
    expect(screen.getByText(testData.title).innerHTML).toContain(testData.title);
  });
});
