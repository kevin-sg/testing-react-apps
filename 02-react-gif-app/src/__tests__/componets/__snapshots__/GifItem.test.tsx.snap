// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifItem/> should do math with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Luffy"
      aria-label="Luffy-image"
      src="https://www.xtrafondos.com/wallpapers/vertical/monkey-d-luffy-de-one-piece-4017.jpg"
    />
    <p>
      Luffy
    </p>
  </div>
</div>
`;
